homeassistant/components/binary_sensor/rainmachine.py

```diff
@@ -0,0 +1,111 @@
+"""
+This platform provides support for binary sensor data from RainMachine.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.rainmachine/
+"""
+
+from logging import getLogger
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.components.rainmachine import (
+    DATA_RAINMACHINE, DATA_UPDATE_TOPIC, RainMachineEntity)
+from homeassistant.const import CONF_MONITORED_CONDITIONS
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+DEPENDENCIES = ['rainmachine']
+
+_LOGGER = getLogger(__name__)
+
+ATTR_FREEZE = 'freeze'
+ATTR_FREEZE_PROTECTION = 'freeze_protection'
+ATTR_HOT_DAYS = 'extra_water_on_hot_days'
+ATTR_HOURLY = 'hourly'
+ATTR_MONTH = 'month'
+ATTR_RAINDELAY = 'raindelay'
+ATTR_RAINSENSOR = 'rainsensor'
+ATTR_WEEKDAY = 'weekday'
+
+SENSORS = {
```

Comment:
No device class ?

Reply:
These are all device configurations, not representations of conditions. Example: `TYPE_FREEZE` means "Is the device currently restricted from watering because it's too cold outside?" That doesn't seem to fit the spirit of the `cold` device class. The same is true for the other sensors.

That said, let me know if you'd rather they be coerced into that format and I'll get it done.